chainer/dataset/tabular/_slice.py

```diff
@@ -32,7 +40,12 @@ def keys(self):
 
     @property
     def mode(self):
-        return self._dataset.mode
+        if self._unary is None:
+            return self._dataset.mode
+        elif self._unary:
+            return None
+        else:
+            return self._dataset.mode or tuple
```

Comment:
I'm just curious but is it valid to index the key against unary slice (e.g. `dataset.slice[:, 'key'].slice[:, ('key',)]`)?

Reply:
I think we don't have such use cases. I just thought supporting that makes the API more simple. Do you prefer raising an exception?

Reply:
No strong preference on it. I just wanted to confirm to what extent the behavior is intended. OK to leave it as is.